[VP][RISCV] Enable promotion on fixed-length vp intrinsics with zvfbfmin

4vtomat opened the pull request:
Following #112393, this aims to promote vp intrinsics for zvfbfmin without zvfbfa

topperc reviewed (approved):
LGTM

lukel97 reviewed (commented):
LGTM but FYI I think long term the plan is that we will eventually remove support for these VP intrinsics in the RISC-V backend: https://discourse.llvm.org/t/rfc-remove-codegen-support-for-trivial-vp-intrinsics-in-the-risc-v-backend/87999

Is this something that you were carrying downstream? Happy to have this patch upstream if it makes merging easier later on

lukel97 reviewed (approved):
LGTM but FYI I think long term the plan is that we will eventually remove support for these VP intrinsics in the RISC-V backend: https://discourse.llvm.org/t/rfc-remove-codegen-support-for-trivial-vp-intrinsics-in-the-risc-v-backend/87999

Is this something that you were carrying downstream? Happy to have this patch upstream if it makes merging easier later on

topperc reviewed (changes requested):
Removing approval until question about VP reductions is answered. VP reductions are not being removed so we do need to support them.

4vtomat committed:
[VP][RISCV] Enable promotion on fixed-length vp intrinsics with zvfbfmin

Following #112393, this aims to promote vp intrinsics for zvfbfmin without zvfbfa
This patch enable all of VP ops in ZvfhminZvfbfminPromoteVPOps but only
add tests for reduce min/max/minimum/maximum since others might be
removed soon:
https://discourse.llvm.org/t/rfc-remove-codegen-support-for-trivial-vp-intrinsics-in-the-risc-v-backend/87999#p-350673-intrinsics-considered-trivial-3

topperc reviewed (approved):
LGTM
